daffodil-lib/src/main/scala/org/apache/daffodil/cookers/EntityReplacer.scala

```diff
@@ -806,17 +816,22 @@ class NonEmptyListOfStringLiteralCharClass_ES_WithByteEntities(pn: String)
 class DelimiterCookerNoES(pn: String) extends ListOfString1OrMoreLiteral(pn, true) {
 
   override val oneLiteralCooker: StringLiteralBase =
-    new StringLiteralNoCharClassEntities(propName, true) {
-
-      override protected def noCharClassEntities(raw: String, context: ThrowsSDE): Unit = {
-        // TODO: this isn't quite right, as it will allow combined delimiters
-        // that still match the empty string, e.g. "%ES;%WSP*;". We could check
-        // if raw.contains("%WSP*;"), but that is too general, preventing valid
-        // delimiters like "foo%WSP*;bar". Although the below matches the
-        // specification, it's probably not the intended behavior.
-        context.schemaDefinitionUnless(raw != "%WSP*;", "Property dfdl:%s cannot contain %%WSP*; when dfdl:lengthKind=\"delimited\".", propName)
-        context.schemaDefinitionUnless(raw != "%ES;", "Property dfdl:%s cannot contain %%ES; when dfdl:lengthKind=\"delimited\".", propName)
+    new StringLiteralNoCharClassEntities(propName, true) with DisallowedCharClassEntitiesMixin {
+
+      //override protected def noCharClassEntities(raw: String, context: ThrowsSDE): Unit = {
+      // TODO: this isn't quite right, as it will allow combined delimiters
+      // that still match the empty string, e.g. "%ES;%WSP*;". We could check
+      // if raw.contains("%WSP*;"), but that is too general, preventing valid
+      // delimiters like "foo%WSP*;bar". Although the below matches the
+      // specification, it's probably not the intended behavior.
+      override val disallowedCharClassEntities = Seq("ES")
+      override def testRaw(raw: String, context: ThrowsSDE) = {
+        context.schemaDefinitionUnless(raw != "%WSP*;", """The WSP* entity cannot appear on it's own when dfdl:lengthKind="delimited".""")
+        super[DisallowedCharClassEntitiesMixin].testRaw(raw, context)
       }
+      //context.schemaDefinitionUnless(raw != "%WSP*;", "Property dfdl:%s cannot contain %%WSP*; when dfdl:lengthKind=\"delimited\".", propName)
+      //context.schemaDefinitionUnless(raw != "%ES;", "Property dfdl:%s cannot contain %%ES; when dfdl:lengthKind=\"delimited\".", propName)
+      //}
     }
 }
 
```

Comment:
I think what you have here for DelimiterCookerNoES is correct for the dfdl separator property, but is not correct for dfdl:terminator. ES is allowed in the dfdl:terminator property, so we need a different cooker that has different checks.

For the dfdl:terminator proeprty, there are no disallowed character class entities, and both WSP* and ES canot appear on their own when lengthKind is delimited. This logic is different from this ``DelimiterCookerNoES``. So you need a new class that is similar to this cooker (maybe call it ``DelimiterCookerNoSoleES``), but it should not mixin the ``DisallowedCharClassEntitiesMixin`` and it should not have ``disallowedCharClassEntities``. All it has is ``testRaw`` that checks for the same WSP* check, but also does the same check for ES.

Then, in Cookers.scala, ``object TerminatorCookerNoES extends DelimiterCookerNoES("terminator")`` extends this new ``DelimiterCokerNoSoleES``.